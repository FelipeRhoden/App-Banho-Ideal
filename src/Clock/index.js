import React from 'react';
import {View, Text} from 'react-native';

export default class Clock extends React.Component {
    constructor(props){
        props = props || {};
        props.timer = props.timer || '';
        super(props);
    }

    render(){
        return(
            <View>
                <Text>{this.props.timer}</Text>
            </View>
        );
    }
} 