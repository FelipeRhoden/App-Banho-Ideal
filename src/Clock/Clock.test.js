import React from 'react';
import renderer from 'react-test-renderer'

import Clock from './';

describe('<Clock />', function(){
    it('Deveria ter 1 props (timer)', ()=>{
        const clock = new Clock();

        expect(clock.props).toEqual((expect.objectContaining({
            timer: expect.anything(),
        })));
    });
    
    it('Deveria renderizar corretamente', ()=>{
        const clock = renderer.create(<Clock />).toJSON();

        expect(clock).toMatchSnapshot();
    })
})