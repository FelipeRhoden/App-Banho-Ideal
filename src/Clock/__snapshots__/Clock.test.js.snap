// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> Deveria renderizar corretamente 1`] = `
<View>
  <Text />
</View>
`;
